data/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -95,4 +102,24 @@ public GenericAppenderFactory setAll(Map<String, String> properties) {
       throw new UncheckedIOException(e);
     }
   }
+
+  @Override
+  public org.apache.iceberg.io.DataWriter<Record> newDataWriter(EncryptedOutputFile file, FileFormat format,
+                                                                StructLike partition) {
+    return new org.apache.iceberg.io.DataWriter<>(
+        newAppender(file.encryptingOutputFile(), format), format,
+        file.encryptingOutputFile().location(), spec, partition, file.keyMetadata());
+  }
+
+  @Override
+  public EqualityDeleteWriter<Record> newEqDeleteWriter(EncryptedOutputFile outputFile, FileFormat format,
+                                                        StructLike partition) {
+    throw new UnsupportedOperationException("Cannot create equality-delete writer for generic record now.");
```

Comment:
Why not implement this?

Reply:
I expected to provide the implemented methods with complete coverage unit tests. But for the purpose providing the PoC solution as soon as possible,  I did't have the time to write those tests so left them unsupported now.